### PR TITLE
Deprecate v1.0 from docs

### PR DESCRIPTION
**Pull request by matthewshaver**
## What are you changing in this pull request and why?

Deprecating dbt Core v1.0 from the docs
- Removing version blocks from the text where firstVersion="1.0"
- Removing version blocks and text body from elements where lastVersion="1.0"
- Removing 1.0 from the version menu

## Checklist

- [x] Add versioning components, as described in [Versioning Docs](https://github.com/dbt-labs/docs.getdbt.com/blob/current/contributing/single-sourcing-content.md#versioning-entire-pages)
- [x] Add a note to the prerelease version [Migration Guide](https://github.com/dbt-labs/docs.getdbt.com/tree/current/website/docs/guides/migration/versions)

- [x] Review the [Content style guide](https://github.com/dbt-labs/docs.getdbt.com/blob/current/contributing/content-style-guide.md) and [About versioning](https://github.com/dbt-labs/docs.getdbt.com/blob/current/contributing/single-sourcing-content.md#adding-a-new-version) so my content adheres to these guidelines.
- [x] Add a checklist item for anything that needs to happen before this PR is merged, such as "needs technical review" or "change base branch."

### CODE REVIEW

File: website/docs/docs/building-a-dbt-project/building-models/python-models.md [hunk 1]
```diff
@@ -19,7 +19,7 @@ Below, you'll see sections entitled "❓ **Our questions**." We are excited to h
 
 dbt Python ("dbt-py") models will help you solve use cases that can't be solved with SQL. You can perform analyses using tools available in the open source Python ecosystem, including state-of-the-art packages for data science and statistics. Before, you would have needed separate infrastructure and orchestration to run Python transformations in production. By defining your Python transformations in dbt, they're just models in your project, with all the same capabilities around testing, documentation, and lineage.
 
-<VersionBlock firstVersion="1.0" lastVersion="1.2">
+<VersionBlock lastVersion="1.2">
```
Comment (runleonarun) [inline website/docs/docs/building-a-dbt-project/building-models/python-models.md]: @matthewshaver this versioning doesn't seem to be working. But it's also not working on docs.getdbt.com either.

Comment (runleonarun) [reply website/docs/docs/building-a-dbt-project/building-models/python-models.md]: Oh wait. is this the wrong/outdated page that was versioned? Should we be versioning https://docs.getdbt.com/docs/build/python-models#overview

File: website/docs/docs/build/incremental-models.md [hunk 2]
```diff
@@ -79,12 +79,6 @@ A `unique_key` enables updating existing rows instead of just appending new rows
 
 Not specifying a `unique_key` will result in append-only behavior, which means dbt inserts all rows returned by the model's SQL into the preexisting target table without regard for whether the rows represent duplicates.
 
-<VersionBlock lastVersion="1.0">
```
Comment (runleonarun) [inline website/docs/docs/build/incremental-models.md:82]: this looks good 👍🏻

**Review by runleonarun — COMMENTED** [review]
Hey Matt! Looks good but some weird versioning going on. I think we might need to delete an old page and version the correct one?

**Review by runleonarun — APPROVED** [review]
Looks good!

File: website/docs/docs/core/connect-data-platform/spark-setup.md [hunk 3]
```diff
@@ -231,13 +229,15 @@ connect_retries: 3
 
 </File>
 
+
 </VersionBlock>
```
Comment (matthewshaver) [inline website/docs/docs/core/connect-data-platform/spark-setup.md]:
```suggestion
```

**Pull request merged**